Make it so no warnings are issues in tests of `plasmapy.particles` (#2314)
PlasmaPy/PlasmaPy: changelog/2314.trivial.1.rst

> Made it so that `plasmapy.particles` no longer emits warnings during
> tests. Doing this required decorating test functions with
> `pytest.mark.filterwarnings` and changing the `pytest` configuration
> so that `~plasmapy.utils.exceptions.PlasmaPyDeprecationWarning` is
> ignored during tests.

Author: ```suggestion
Reduced the number of warnings emitted by `plasmapy.particles` during
tests by decorating test functions with `pytest.mark.filterwarnings`.
```
